### PR TITLE
DataDog Service Catalog Meta Data now in source control and updated with PR merge

### PR DESCRIPTION
**Pull request by lexicalninja**
## Summary

This change adds a github action that updates the datadog service catalog from a single file anytime the service metadata in the `datadog-service-catalog.yml` is updated. 

This allows for a single source of truth of all the DataDog service catalog metadata.

*This action runs on PR merges to main that update the `datadog-service-catalog.yml`*

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/68112

## Testing done
This has been tested and successfully runs in a local act environment. 

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x] Export service catalog entries as YAML
- [x] Add YAML to vets-api source control
- [x] Create a programmatic mechanism for entries to be pushed to Datadog when updated (e.g. via Github Action on merges to main)

## Requested Feedback
Please review this draft and provide any feedback or ask any questions. I have made this extremely basic and straightforward in this first pass so that we can uncover any issues related to process, etc. 

I will also need to get 2 secrets added to the vets-api gh actions environtment.

### CODE REVIEW

File: datadog-service-catalog.yml
Comment (patrickvinograd) [inline datadog-service-catalog.yml:151]: let's wipe this test-10-10-cg service - it was something Aparna was trying, as a mapping from what's in the VFS application directory. But I don't like having individuals emails in there and it conflicts with the `caregiver-application` service.

File: .github/workflows/datadog_service_catalog_update.yml
Comment (patrickvinograd) [inline .github/workflows/datadog_service_catalog_update.yml:1]: This ought to get updated before merging for real

**Review by patrickvinograd — COMMENTED** [review]
Overall this looks simple enough to be a good starting point. 

* Wondering what happens if somebody manually edits and creates some YAML that is altogether invalid, or just doesn't conform to the datadog yaml schema. Should we run `yq v -d '*'` (validate all docs) on pull requests? That at least checks for valid yaml. 
* If I had my druthers these entries would be in alphabetical order but it seems unlikely that state would persist for very long even if I asked you to do it here and now.

**Comment by acrollet** [comment]
> * Wondering what happens if somebody manually edits and creates some YAML that is altogether invalid, or just doesn't conform to the datadog yaml schema. Should we run `yq v -d '*'` (validate all docs) on pull requests? That at least checks for valid yaml.

Even better, could we go ahead and use something like https://github.com/marketplace/actions/validate-json-schemas with https://github.com/DataDog/schema/blob/main/service-catalog/v2.1/schema.json in a separate action to validate the schema?

**Comment by acrollet** [comment]
> * If I had my druthers these entries would be in alphabetical order but it seems unlikely that state would persist for very long even if I asked you to do it here and now.

agree with the desire, and with it being a losing battle

**Comment by lexicalninja** [comment]
> > * If I had my druthers these entries would be in alphabetical order but it seems unlikely that state would persist for very long even if I asked you to do it here and now.
> 
> agree with the desire, and with it being a losing battle

we can make each team put this in their root and then do a path for `'**/datadog-service-catalog.yml'` or whatever later on if it gets unweildy this way. It would take more work than this is doing, but its a possibility if this turns out to be useful

**Comment by lexicalninja** [comment]
> > > > * If I had my druthers these entries would be in alphabetical order but it seems unlikely that state would persist for very long even if I asked you to do it here and now.
> > > 
> > > 
> > > agree with the desire, and with it being a losing battle
> > 
> > 
> > we can make each team put this in their root and then do a path for `'**/datadog-service-catalog.yml'` or whatever later on if it gets unweildy this way. It would take more work than this is doing, but its a possibility if this turns out to be useful
> 
> I had thought about this approach and really like it in the abstract, but I think that:
> 
> * it runs into some issues with services that don't have a corresponding vets-api module/controller
> * shipping the simpler thing first is good

😆 I was definitely shipping this first. I meant after we see if this way of controlling in source code works and that was a future option we could noodle on

**Comment by lexicalninja** [comment]
> > * Wondering what happens if somebody manually edits and creates some YAML that is altogether invalid, or just doesn't conform to the datadog yaml schema. Should we run `yq v -d '*'` (validate all docs) on pull requests? That at least checks for valid yaml.
> 
> Even better, could we go ahead and use something like https://github.com/marketplace/actions/validate-json-schemas with https://github.com/DataDog/schema/blob/main/service-catalog/v2.1/schema.json in a separate action to validate the schema?

This validator doesnt like yaml files. ill see if there is another way to validate yaml against a json schema.

One thing I did was update a service with bad schema attributes. the api updated the service, sent back a warning about the attr not being allowed, but the attribute is in [dd in the service definition](https://vagov.ddog-gov.com/services?currentTab=Definition&env=%2A&lens=Performance&page=2&selectedService=test-foo-too&start=1698849123951&end=1698852723951).

**Comment by acrollet** [comment]
> This validator doesnt like yaml files. ill see if there is another way to validate yaml against a json schema.

It looks like this tool can do it: https://github.com/json-schema-everywhere/pajv - not sure if there's an existing gh action that wraps it.
 
> One thing I did was update a service with bad schema attributes. the api updated the service, sent back a warning about the attr not being allowed, but the attribute is in [dd in the service definition](https://vagov.ddog-gov.com/services?currentTab=Definition&env=%2A&lens=Performance&page=2&selectedService=test-foo-too&start=1698849123951&end=1698852723951).

interesting that they allow it 🤔

**Comment by patrickvinograd** [comment]
What about separate files, but in a single subdirectory? Then `ls` gives us our sort capability, and we avoid being nagged about lines of code limits.

**Comment by lexicalninja** [comment]
What team should be assigned as codeowners for the datadog service files? @acrollet @patrickvinograd 
I am going to merge this with one file in the folder for now to make it simple and get the next PR up for the validation work and then we can go back and make this work for multiple docs

**Comment by acrollet** [comment]
> What team should be assigned as codeowners for the datadog service files? @acrollet @patrickvinograd I am going to merge this with one file in the folder for now to make it simple and get the next PR up for the validation work and then we can go back and make this work for multiple docs

I think I would just go with `@department-of-veterans-affairs/backend-review-group` for now, I don't see anything clearly more logical and it can always be changed later if we need to make a new group. Your plan sounds good 🙂

**Review by acrollet — APPROVED** [review]

**Pull request merged**
